pkg/preflight/aws.go

```diff
@@ -55,6 +73,45 @@ func (a *aws) initCloudOps() error {
 	return nil
 }
 
+func (a *aws) setAWSCredentialEnvVars(cluster *corev1.StorageCluster) error {
+	// Check if user provided credentials exist
+	var accessKeyEnv, secretKeyEnv *v1.EnvVar
+	for _, env := range cluster.Spec.Env {
+		if env.Name == awsAccessKeyEnvName {
+			accessKeyEnv = env.DeepCopy()
+		} else if env.Name == awsSecretKeyEnvName {
+			secretKeyEnv = env.DeepCopy()
+		}
+	}
+	if accessKeyEnv == nil && secretKeyEnv == nil {
+		logrus.Debugf("no aws credentials provided, will use instance privileges instead")
+		return nil
+	} else if accessKeyEnv == nil || secretKeyEnv == nil {
+		return fmt.Errorf("both AWS_ACCESS_KEY_ID and AWS_ACCESS_KEY_ID need to be provided")
+	}
+
+	// Get the credentials from secrets or values
+	accessKey, err := pxutil.GetValueFromEnvVar(context.TODO(), a.k8sClient, accessKeyEnv, cluster.Namespace)
+	if err != nil {
+		return err
+	}
+	secretKey, err := pxutil.GetValueFromEnvVar(context.TODO(), a.k8sClient, secretKeyEnv, cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Set environment variables using credentials above
+	if err := os.Setenv(awsAccessKeyEnvName, accessKey); err != nil {
+		return err
+	}
+	if err := os.Setenv(awsSecretKeyEnvName, secretKey); err != nil {
```

Comment:
is this a new or legacy behavior? is there any security concern about exposing it in env var?

Reply:
@haibinxie based on doc https://docs.portworx.com/install-portworx/cloud/aws/aws-asg/
we can either grant the ec2 instance privileges or pass this user specified env vars to storage cluster spec, then the credentials will be passed to px pods.
the behavior has been there. If we are passing it to px pod, I don't see security concerns here.

Reply:
We should not be setting these values in the operator's memory. We do pass the values to the pods but not directly. We use the `env.valueFrom` field and don't directly add the value ourselves. Kubernetes takes the responsibility of fetching the secret and mounting it in the pod securely.

Reply:
but this is operator pod, we cannot edit operator pod to add volumes and mounts. we can use stc.env.valueFrom to get the credential from storage cluster, then do we have other better ways to set this env in operator container? @piyush-nimbalkar

Reply:
My worry is that if someone gets access to the operator pod, they can exec into it and print env values.

I was hoping we will get the values from the secret and directly pass them to the AWS sdk client directly. But then I just saw that we are using the cloudops package which expects the creds in the env variables.
I don't see any other way apart from adding a new NewClient to cloudops to take the id and secret as parameters. Let's merge this change for now and explore adding a new method to cloudops.

Reply:
created https://portworx.atlassian.net/browse/PWX-28664 to track